### PR TITLE
chore: Switch from flake8+isort to ruff

### PR DESCRIPTION
**Pull request by lukapeschke**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,15 +10,10 @@ repos:
     rev: 22.8.0
     hooks:
     -   id: black
--   repo: http://github.com/PyCQA/isort
-    rev: 5.10.1
+-   repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.137
```
Comment (PrettyWood) [inline .pre-commit-config.yaml:14]: we have ruff>=0.0.138,<0.1 in `test-requirements`

Comment (lukapeschke) [reply .pre-commit-config.yaml:14]: Yeah but there's no `v0.0.138` tag on the pre-commit repo yet :confused:

Comment (PrettyWood) [reply .pre-commit-config.yaml:14]: I'd rather use the local ruff version
```
  - id: lint
    name: lint
    entry: make lint
    types: [python]
    language: system
    pass_filenames: false
```

Comment (lukapeschke) [reply .pre-commit-config.yaml:14]: Ah sweet, didn't know this was possible with pre-commit :+1: Will update!

Comment (lukapeschke) [reply .pre-commit-config.yaml]: @PrettyWood Done, for both python and rust. For now it's fast enough, so it runs on everything, anytime a rust or python file changes. If it gets slow in the future, I'll split the `make` targets per language

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -6,21 +6,21 @@ repos:
...
--   repo: http://github.com/PyCQA/flake8
-    rev: 5.0.4
-    hooks:
-    -   id: flake8
-        exclude: test.py
 -   repo: http://github.com/doublify/pre-commit-rust
     rev: v1.0
     hooks:
-    -   id: fmt
     -   id: cargo-check
```
Comment (PrettyWood) [inline .pre-commit-config.yaml:12]: do we need this?

Comment (lukapeschke) [reply .pre-commit-config.yaml:12]: Well I can remove it, but we'd need a `make` target for `cargo check`

**Review by PrettyWood — APPROVED** [review]
Great thanks

**Pull request merged**